ntoskrnl/ex/sysinfo.c

```diff
@@ -2503,6 +2568,109 @@ QSI_DEF(SystemExtendedHandleInformation)
     return Status;
 }
 
+/* Class 76 - System firmware table information  */
+QSI_DEF(SystemFirmwareTableInformation)
+{
+    PSYSTEM_FIRMWARE_TABLE_INFORMATION SysFirmwareInfo = (PSYSTEM_FIRMWARE_TABLE_INFORMATION)Buffer;
+    NTSTATUS Status = STATUS_SUCCESS;
+    ULONG DataSize = 0;
+    ULONG TableCount = 0;
+
+    DPRINT("NtQuerySystemInformation - SystemFirmwareTableInformation\n");
+
+    /* Set initial required buffer size */
+    *ReqSize = FIELD_OFFSET(SYSTEM_FIRMWARE_TABLE_INFORMATION, TableBuffer);
+
+    /* Check user's buffer size */
+    if (Size < *ReqSize)
+    {
+        return STATUS_INFO_LENGTH_MISMATCH;
+    }
+
+    switch (SysFirmwareInfo->ProviderSignature)
+    {
+        /*
+         * ExpFirmwareTableResource and ExpFirmwareTableProviderListHead
+         * variables should be used there somehow...
```

Comment:
Which means, they are loaded and initialized at boot and cached in memory, and then just read later... ?

Reply:
A provider should register itself using the related `SystemRegisterFirmwareTableInformationHandler` class, then it will probably be added to this list.
After that, no clue yet :D

Reply:
Yeah, these things are not implemented yet, so the comment is for the future :)

Reply:
Add a `FIXME: `?

Reply:
The class already have DPRINTs that notify about missing unimplemented things.